ros2_batch_job/__main__.py

```diff
@@ -441,7 +447,52 @@ def run(args, build_function, blacklisted_package_names=None):
         # Print the pip version
         job.run(['"%s"' % job.python, '-m', 'pip', '--version'], shell=True)
         # Install pip dependencies
-        job.run(['"%s"' % job.python, '-m', 'pip', 'install', '-U'] + pip_dependencies, shell=True)
+        install_pip_packages = list(pip_dependencies)
```

Comment:
this variable name sounds like a command more than a package list.
We should renaming to something like `pip_packages_to_install`

Reply:
Why is that temporary variable needed in the first place ? Isn't `pip_dependencies` already a  list that we can extend?
Would something like that do the job?
```
if not args.colcon_branch:
    pip_dependencies.extend(colcon_packages)
```

Reply:
`pip_dependencies` is a global variable. Altering its content within this function doesn't sounds like a good idea to me.

Renamed to `pip_packages` in 5d377b8cefa7cb951a6bd28ab5b309a8c7b6f84e.